ThirdpartyToolchain.cmake

```diff
@@ -0,0 +1,205 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# MODULE:   ThirdpartyToolchain
+#
+# PROVIDES:
+#   resolve_dependency( DEPENDENCY_NAME dependencyName
+#                    [REQUIRED_VERSION required version]
+#                    ...
+#   )
+#
+#       Provides the ability to resolve third party dependencies. If the dependency is
+#       already available in the system it will be used.
+#
+#       The DEPENDENCY_NAME argument is required. The dependencyName value will be used to search for
+#       the installed dependencies Config file and thus this name should match find_package()
+#       standards.
+#
+# EXAMPLE USAGE:
+#   # Download and setup or use already installed dependency.
+#   include(ThirdpartyToolchain)
+#   resolve_dependency(folly)
+#
+#
+#========================================================================================
+
+
+add_custom_target(toolchain)
+include(ExternalProject)
+
+#=====================================FOLLY==============================================
+
+if(DEFINED ENV{VELOX_FOLLY_URL})
+  set(FOLLY_SOURCE_URL "$ENV{VELOX_FOLLY_URL}")
+else()
+  set(VELOX_FOLLY_BUILD_VERSION v2022.07.11.00)
```

Comment:
Nit: We can probably do this in another PR, but would be nice if we have all our dependencies defined in another file (akin to how python dependencies are configured) and pull that in.